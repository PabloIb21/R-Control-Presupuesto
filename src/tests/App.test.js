import React from 'react';
import { mount } from 'enzyme';
import App from '../App';

describe('AppPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App />);
  });

  test('mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('the application should be displayed after defining the budget', () => {
    wrapper.find('input').at(0).simulate('change', { target: { value: '1000' } });
    wrapper.find('form').simulate('submit');
    expect(wrapper.find('.contenido-principal').exists()).toBe(true);
  });
});