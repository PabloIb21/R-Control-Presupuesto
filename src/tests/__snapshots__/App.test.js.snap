// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPage mount 1`] = `
<div
  className="container"
>
  <header>
    <h1>
      Gasto Semanal
    </h1>
    <div
      className="contenido-principal contenido"
    >
      Array [
        <h2>
          Coloca tu presupuesto
        </h2>,
        <form
          onSubmit={[Function]}
        >
          <input
            className="u-full-width"
            onChange={[Function]}
            placeholder="Coloca tu presupuesto"
            type="number"
          />
          <input
            className="button-primary u-full-width"
            type="submit"
            value="Definir presupuesto"
          />
        </form>,
      ]
    </div>
  </header>
</div>
`;
